text/0000-mem-replace-with.md

```diff
@@ -0,0 +1,64 @@
+- Feature Name: mem-replace-with
+- Start Date: 2016-09-01
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add a new function to `core::mem` (and thus `std::mem`), `replace_with`, which
+invokes a closure that takes temporary ownership over some value behind a
+mutable reference.
+
+This is similar to the functions provided by [the take_mut crate](https://crates.io/crates/take_mut).
+
+# Motivation
+[motivation]: #motivation
+
+A common pattern, especially for low-level programmers, is to acquire ownership
+of a value behind a reference without immediately giving it a replacement value.
+
+`core::mem::replace_with` allows such thing by generalizing
+`core::mem::replace`, to allow the replacement value being generated by a
+closure and even depend on the old value.
+
+Unfortunately, there are no safe implementation in the standard library,
```

Comment:
s/are/is/ to match "implementation".  Or alternatively, "the standard library doesn't provide any safe implementation"